README.md

# Asmaul Husna Flutter Package

A Flutter package that provides access to the 99 Names of Allah (Asmaul Husna) in English and Arabic. This package allows you to retrieve the English name, Arabic name, and a brief description for each of the 99 Names.


## Installation

To use this package, add it to your pubspec.yaml:

```bash
  dependencies:
  asmaulhusna: ^1.0.0

```

    
## Usage/Examples Dart

Import the asmaulhusna library and use the provided functions to access the Names of Allah.

```javascript
import 'package:asmaulhusna/asmaulhusna.dart';

// Get the English name of a specific number (1 to 99)
String englishName = getEnglishName(1);

// Get the Arabic name of a specific number (1 to 99)
String arabicName = getArabicName(1);

// Get the description of a name in English for a specific number (1 to 99)
String description = getDescription(1);

// Get a list of all English names
List<String> allEnglishNames = getEveryEnglishName();

// Get a list of all Arabic names
List<String> allArabicNames = getEveryArabicName();

```

## Usage/Examples Flutter

Import the asmaulhusna library and use the provided functions to access the Names of Allah.

```javascript
import 'package:asmaulhusna/asmaulhusna.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
       
        primarySwatch: Colors.blue,
      ),
      home: const HomePage(),
    );
  }
}

class HomePage extends StatefulWidget {
  const HomePage({super.key});

  @override
  State<HomePage> createState() => HomePageState();
}

List<Text> nintynineNamesInEnglish() {
  List<Text> names = [];
  int i = 1;
  getEveryEnglishName().forEach((element) {
    names.add(Text('${i.toString()} - $element'));
    i++;
  });
  return names;
}

class HomePageState extends State<HomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
          child: Column(
        mainAxisSize: MainAxisSize.min,
        children: [
          Text(getArabicName(1)),
          Text(getEnglishName(1)),
          Text(getDescritption(1)),
          const SizedBox(
            height: 10,
          ),
          const Text('99 names'),
          const SizedBox(
            height: 10,
          ),
          Container(
              color: const Color.fromARGB(255, 231, 225, 166),
              height: 500,
              width: MediaQuery.of(context).size.width,
              child: SingleChildScrollView(
                  child: Column(
                children: [...nintynineNamesInEnglish()],
              )))
        ],
      )),
    );
  }
}

```
![image](https://github.com/aslahmogral/asmaul_husna/assets/68979861/45165fa6-bc48-4815-85af-725ad3c6e331)



## License

This package is released under the [MIT LICENSE](https://github.com/aslahmogral/asmaul_husna/blob/main/LICENSE). Feel free to use and modify it in your projects.

If you have any questions, issues, or suggestions, please open an issue on the GitHub repository.




## Author

This package is maintained by [@aslahmogral](https://github.com/aslahmogral).

Thank you for using the Asmaul Husna Flutter Package!






CHANGELOG.md
## 0.0.2

* Arabic name - added name in arabic

## 0.0.1

* added 99 english name and description


test/asmaulhusna_test.dart
import 'package:flutter_test/flutter_test.dart';

import 'package:asmaulhusna/asmaulhusna.dart';

void main() {
  test('english name', () {
    expect(getEnglishName(28), 'Al-Hakim');
  });

  test('arabic name', () {
    expect(getArabicName(1), 'الرَّحْمَنُ');
  });

  test('asmaul husna descriptoin', () {
    expect(getDescritption(33), 'The Magnificent');
  });
}


lib/asmaulhusna.dart
library asmaulhusna;

import 'package:asmaulhusna/asmaul_husna_text.dart';

/// get english name.
String getEnglishName(int number) {
  return asmaulHusnaText[number]['name_english'];
}

/// get arabic name.
String getArabicName(int number) {
  return asmaulHusnaText[number]['name_arabic'];
}


/// get description of name in english .
String getDescritption(int number) {
  return asmaulHusnaText[number]['desc_english'];
}


//get every english name
List<String> getEveryEnglishName() {
  List<String> names = [];
  asmaulHusnaText.forEach((key, value) {
    names.add(asmaulHusnaText[key]['name_english']);
  });
  return names;
}

List<String> getEveryArabicName() {
  List<String> names = [];
  asmaulHusnaText.forEach((key, value) {
    names.add(asmaulHusnaText[key]['name_arabic']);
  });
  return names;
}



lib/asmaul_husna_text.dart
const Map asmaulHusnaText= {
  1: {
    'name_arabic': 'ٱلْرَّحْمَـانُ',
    'name_english': 'Ar-Rahman',
    'desc_english': 'The Most Gracious',
  },
  2: {
    'name_arabic': 'ٱلْرَّحِيْمُ',
    'name_english': 'Ar-Raheem',
    'desc_english': 'The Most Merciful',
  },
  3: {
    'name_arabic': 'ٱلْمَلِكُ',
    'name_english': 'Al-Malik',
    'desc_english': 'The King',
  },
  4: {
    'name_arabic': 'ٱلْقُدُّوسُ',
    'name_english': 'Al-Quddus',
    'desc_english': 'The Most Holy',
  },
  5: {
    'name_arabic': 'ٱلْسَّلَامُ',
    'name_english': 'As-Salaam',
    'desc_english': 'The Source of Peace',
  },
  6: {
    'name_arabic': 'ٱلْمُؤْمِنُ',
    'name_english': 'Al-Mu min',
    'desc_english': 'The Guardian of Faith',
  },
  7: {
    'name_arabic': 'ٱلْمُهَيْمِنُ',
    'name_english': 'Al-Muhaymin',
    'desc_english': 'The Protector',
  },
  8: {
    'name_arabic': 'ٱلْعَزِيزُ',
    'name_english': 'Al-Aziz',
    'desc_english': 'The Mighty',
  },
  9: {
    'name_arabic': 'ٱلْجَبَّارُ',
    'name_english': 'Al-Jabbar',
    'desc_english': 'The Compeller',
  },
  10: {
    'name_arabic': 'ٱلْمُتَكَبِّرُ',
    'name_english': 'Al-Mutakabbir',
    'desc_english': 'The Majestic',
  },
  11: {
    'name_arabic': 'ٱلْخَالِقُ',
    'name_english': 'Al-Khaliq',
    'desc_english': 'The Creator',
  },
  12: {
    'name_arabic': 'ٱلْبَارِئُ',
    'name_english': 'Al-Bari',
    'desc_english': 'The Evolver',
  },
  13: {
    'name_arabic': 'ٱلْمُصَوِّرُ',
    'name_english': 'Al-Musawwir',
    'desc_english': 'The Fashioner of Forms',
  },
  14: {
    'name_arabic': 'ٱلْغَفَّارُ',
    'name_english': 'Al-Ghaffar',
    'desc_english': 'The Forgiving',
  },
  15: {
    'name_arabic': 'ٱلْقَهَّارُ',
    'name_english': 'Al-Qahhar',
    'desc_english': 'The Subduer',
  },
  16: {
    'name_arabic': 'ٱلْوَهَّابُ',
    'name_english': 'Al-Wahhab',
    'desc_english': 'The Bestower',
  },
  17: {
    'name_arabic': 'ٱلْرَّزَّاقُ',
    'name_english': 'Ar-Razzaq',
    'desc_english': 'The Provider',
  },
  18: {
    'name_arabic': 'ٱلْفَتَّاحُ',
    'name_english': 'Al-Fattah',
    'desc_english': 'The Opener',
  },
  19: {
    'name_arabic': 'ٱلْعَلِيمُ',
    'name_english': 'Al-Alim',
    'desc_english': 'The Knower of All',
  },
  20: {
    'name_arabic': 'ٱلْقَابِضُ',
    'name_english': 'Al-Qabid',
    'desc_english': 'The Constrictor',
  },
  21: {
    'name_arabic': 'ٱلْبَاسِطُ',
    'name_english': 'Al-Basit',
    'desc_english': 'The Expander',
  },
  22: {
    'name_arabic': 'ٱلْخَافِضُ',
    'name_english': 'Al-Khafid',
    'desc_english': 'The Abaser',
  },
  23: {
    'name_arabic': 'ٱلْرَّافِعُ',
    'name_english': 'Ar-Rafi',
    'desc_english': 'The Exalter',
  },
  24: {
    'name_arabic': 'ٱلْمُعِزُّ',
    'name_english': 'Al-Mu\'izz',
    'desc_english': 'The Bestower of Honor',
  },
  25: {
    'name_arabic': 'ٱلْمُذِلُّ',
    'name_english': 'Al-Mudhill',
    'desc_english': 'The Humiliator',
  },
  26: {
    'name_arabic': 'ٱلْسَّمِيعُ',
    'name_english': 'As-Sami',
    'desc_english': 'The Hearer of All',
  },
  27: {
    'name_arabic': 'ٱلْبَصِيرُ',
    'name_english': 'Al-Basir',
    'desc_english': 'The Seer of All',
  },
  28: {
    'name_arabic': 'ٱلْحَكَمُ',
    'name_english': 'Al-Hakim',
    'desc_english': 'The Perfectly Wise',
  },
  29: {
    'name_arabic': 'ٱلْعَدْلُ',
    'name_english': 'Al-Adl',
    'desc_english': 'The Just',
  },
  30: {
    'name_arabic': 'ٱلْلَّطِيفُ',
    'name_english': 'Al-Latif',
    'desc_english': 'The Subtle',
  },
  31: {
    'name_arabic': 'ٱلْخَبِيرُ',
    'name_english': 'Al-Khabir',
    'desc_english': 'The All-Aware',
  },
  32: {
    'name_arabic': 'ٱلْحَلِيمُ',
    'name_english': 'Al-Halim',
    'desc_english': 'The Forbearing',
  },
  33: {
    'name_arabic': 'ٱلْعَظِيمُ',
    'name_english': 'Al-Azim',
    'desc_english': 'The Magnificent',
  },
  34: {
    'name_arabic': 'ٱلْغَفُورُ',
    'name_english': 'Al-Ghafur',
    'desc_english': 'The Forgiver and Concealer of Faults',
  },
  35: {
    'name_arabic': 'ٱلْشَّكُورُ',
    'name_english': 'Ash-Shakur',
    'desc_english': 'The Rewarder of Thankfulness',
  },
  36: {
    'name_arabic': 'ٱلْعَلِيُّ',
    'name_english': 'Al-Ali',
    'desc_english': 'The Most High',
  },
  37: {
    'name_arabic': 'ٱلْكَبِيرُ',
    'name_english': 'Al-Kabir',
    'desc_english': 'The Most Great',
  },
  38: {
    'name_arabic': 'ٱلْحَفِيظُ',
    'name_english': 'Al-Hafiz',
    'desc_english': 'The Preserver',
  },
  39: {
    'name_arabic': 'ٱلْمُقِيتُ',
    'name_english': 'Al-Muqit',
    'desc_english': 'The Nourisher',
  },
  40: {
    'name_arabic': 'ٱلْحَسِيبُ',
    'name_english': 'Al-Hasib',
    'desc_english': 'The Accounter',
  },
  41: {
    'name_arabic': 'ٱلْجَلِيلُ',
    'name_english': 'Aj-Jalil',
    'desc_english': 'The Mighty',
  },
  42: {
    'name_arabic': 'ٱلْكَرِيمُ',
    'name_english': 'Al-Karim',
    'desc_english': 'The Generous',
  },
  43: {
    'name_arabic': 'ٱلْرَّقِيبُ',
    'name_english': 'Ar-Raqib',
    'desc_english': 'The Watchful',
  },
  44: {
    'name_arabic': 'ٱلْمُجِيبُ',
    'name_english': 'Al-Mujib',
    'desc_english': 'The Responder to Prayer',
  },
  45: {
    'name_arabic': 'ٱلْوَاسِعُ',
    'name_english': 'Al-Wasi',
    'desc_english': 'The All-Comprehending',
  },
  46: {
    'name_arabic': 'ٱلْحَكِيمُ',
    'name_english': 'Al-Hakim',
    'desc_english': 'The Perfectly Wise',
  },
  47: {
    'name_arabic': 'ٱلْوَدُودُ',
    'name_english': 'Al-Wadud',
    'desc_english': 'The Loving',
  },
  48: {
    'name_arabic': 'ٱلْمَجِيدُ',
    'name_english': 'Al-Majid',
    'desc_english': 'The Majestic',
  },
  49: {
    'name_arabic': 'ٱلْبَاعِثُ',
    'name_english': 'Al-Ba\'ith',
    'desc_english': 'The Resurrector',
  },
  50: {
    'name_arabic': 'ٱلْشَّهِيدُ',
    'name_english': 'Ash-Shahid',
    'desc_english': 'The Witness',
  },

  51: {
    'name_arabic': 'ٱلْحَقُّ',
    'name_english': 'Al-Haqq',
    'desc_english': 'The Truth',
  },
  52: {
    'name_arabic': 'ٱلْوَكِيلُ',
    'name_english': 'Al-Wakil',
    'desc_english': 'The Trustee',
  },
  53: {
    'name_arabic': 'ٱلْقَوِيُّ',
    'name_english': 'Al-Qawi',
    'desc_english': 'The Strong',
  },
  54: {
    'name_arabic': 'ٱلْمَتِينُ',
    'name_english': 'Al-Matin',
    'desc_english': 'The Firm',
  },
  55: {
    'name_arabic': 'ٱلْوَلِيُّ',
    'name_english': 'Al-Wali',
    'desc_english': 'The Friend',
  },
  56: {
    'name_arabic': 'ٱلْحَمِيدُ',
    'name_english': 'Al-Hamid',
    'desc_english': 'The Praiseworthy',
  },
  57: {
    'name_arabic': 'ٱلْمُحْصِيُ',
    'name_english': 'Al-Muhsi',
    'desc_english': 'The Counter',
  },
  58: {
    'name_arabic': 'ٱلْمُبْدِئُ',
    'name_english': 'Al-Mubdi',
    'desc_english': 'The Originator',
  },
  59: {
    'name_arabic': 'ٱلْمُعِيدُ',
    'name_english': 'Al-Mu\'id',
    'desc_english': 'The Restorer',
  },
  60: {
    'name_arabic': 'ٱلْمُحْيِى',
    'name_english': 'Al-Muhyi',
    'desc_english': 'The Giver of Life',
  },
  61: {
    'name_arabic': 'ٱلْمُمِيتُ',
    'name_english': 'Al-Mumit',
    'desc_english': 'The Taker of Life',
  },
  62: {
    'name_arabic': 'ٱلْحَىُّ',
    'name_english': 'Al-Hayy',
    'desc_english': 'The Ever-Living',
  },
  63: {
    'name_arabic': 'ٱلْقَيُّومُ',
    'name_english': 'Al-Qayyum',
    'desc_english': 'The Sustainer',
  },
  64: {
    'name_arabic': 'ٱلْوَاجِدُ',
    'name_english': 'Al-Wajid',
    'desc_english': 'The Finder',
  },
  65: {
    'name_arabic': 'ٱلْمَاجِدُ',
    'name_english': 'Al-Majid',
    'desc_english': 'The Glorious',
  },
  66: {
    'name_arabic': 'ٱلْوَاحِدُ',
    'name_english': 'Al-Wahid',
    'desc_english': 'The One',
  },
  67: {
    'name_arabic': 'ٱلْأَحَد',
    'name_english': 'AL-AHAD',
    'desc_english': 'The Unique, The Only One',
  },
  68: {
    'name_arabic': 'ٱلْصَّمَدُ',
    'name_english': 'AS-SAMAD',
    'desc_english': 'The Eternal, Satisfier of Needs',
  },
  69: {
    'name_arabic': 'ٱلْقَادِرُ',
    'name_english': 'Al-Qadir',
    'desc_english': 'The Powerful',
  },
  70: {
    'name_arabic': 'ٱلْمُقْتَدِرُ',
    'name_english': 'Al-Muqtadir',
    'desc_english': 'The Omnipotent',
  },
  71: {
    'name_arabic': 'ٱلْمُقَدِّمُ',
    'name_english': 'Al-Muqaddim',
    'desc_english': 'The Expeditiously All-Arranger',
  },
  72: {
    'name_arabic': 'ٱلْمُؤَخِّرُ',
    'name_english': 'Al-Mu\'akhkhir',
    'desc_english': 'The Delayer',
  },
  73: {
    'name_arabic': 'ٱلأَوَّلُ',
    'name_english': 'Al-Awwal',
    'desc_english': 'The First',
  },
  74: {
    'name_arabic': 'ٱلْآخِرُ',
    'name_english': 'Al-Akhir',
    'desc_english': 'The Last',
  },
  75: {
    'name_arabic': 'ٱلْظَّاهِرُ',
    'name_english': 'Az-Zahir',
    'desc_english': 'The Manifest',
  },
  76: {
    'name_arabic': 'ٱلْبَاطِنُ',
    'name_english': 'Al-Batin',
    'desc_english': 'The Hidden',
  },
  77: {
    'name_arabic': 'ٱلْوَالِي',
    'name_english': 'Al-Wali',
    'desc_english': 'The Protecting Friend',
  },
  78: {
    'name_arabic': 'ٱلْمُتَعَالِي',
    'name_english': 'Al-Muta\'ali',
    'desc_english': 'The Supreme',
  },
  79: {
    'name_arabic': 'ٱلْبَرُّ',
    'name_english': 'Al-Barr',
    'desc_english': 'The Doer of Good',
  },
  80: {
    'name_arabic': 'ٱلْتَّوَّابُ',
    'name_english': 'At-Tawwab',
    'desc_english': 'The Acceptor of Repentance',
  },
  81: {
    'name_arabic': 'ٱلْمُنْتَقِمُ',
    'name_english': 'Al-Muntaqim',
    'desc_english': 'The Avenger',
  },
  82: {
    'name_arabic': 'ٱلْعَفُوُّ',
    'name_english': 'Al-Afuw',
    'desc_english': 'The Pardoner',
  },
  83: {
    'name_arabic': 'ٱلْرَّؤُفُ',
    'name_english': 'Ar-Ra\'uf',
    'desc_english': 'The Compassionate',
  },
  84: {
    'name_arabic': 'مَالِكُ ٱلْمُلْكُ	',
    'name_english': 'Malik-ul-Mulk',
    'desc_english': 'The Owner of All Sovereignty',
  },
  85: {
    'name_arabic': 'ذُو الْجَلاَلِ وَالإِكْرَامِ',
    'name_english': 'Dhul-Jalali wal-Ikram',
    'desc_english': 'The Lord of Majesty and Generosity',
  },
  86: {
    'name_arabic': 'ٱلْمُقْسِطُ',
    'name_english': 'Al-Muqsit',
    'desc_english': 'The Equitable',
  },
  87: {
    'name_arabic': 'ٱلْجَامِعُ',
    'name_english': 'Al-Jami',
    'desc_english': 'The Gatherer',
  },
  88: {
    'name_arabic': 'ٱلْغَنيُّ',
    'name_english': 'Al-Ghani',
    'desc_english': 'The Self-Sufficient',
  },
  89: {
    'name_arabic': 'ٱلْمُغْنِيُّ',
    'name_english': 'Al-Mughni',
    'desc_english': 'The Enricher',
  },
  90: {
    'name_arabic': 'ٱلْمَانِعُ',
    'name_english': 'Al-Mani',
    'desc_english': 'The Preventer of Harm',
  },
  91: {
    'name_arabic': 'ٱلْضَّارُ',
    'name_english': 'Ad-Darr',
    'desc_english': 'The Creator of Harmful',
  },
  92: {
    'name_arabic': 'ٱلْنَّافِعُ',
    'name_english': 'An-Nafi',
    'desc_english': 'The Creator of Good',
  },
  93: {
    'name_arabic': 'ٱلْنُّورُ',
    'name_english': 'An-Nur',
    'desc_english': 'The Light',
  },
  94: {
    'name_arabic': 'ٱلْهَادِي',
    'name_english': 'Al-Hadi',
    'desc_english': 'The Guide',
  },
  95: {
    'name_arabic': 'ٱلْبَدِيعُ',
    'name_english': 'Al-Badi',
    'desc_english': 'The Incomparable',
  },
  96: {
    'name_arabic': 'ٱلْبَاقِي',
    'name_english': 'Al-Baqi',
    'desc_english': 'The Lasting',
  },
  97: {
    'name_arabic': 'ٱلْوَارِثُ',
    'name_english': 'Al-Warith',
    'desc_english': 'The Inheritor of All',
  },
  98: {
    'name_arabic': 'ٱلْرَّشِيدُ',
    'name_english': 'Ar-Rashid',
    'desc_english': 'The Righteous Teacher',
  },
  99: {
    'name_arabic': 'ٱلْصَّبُورُ',
    'name_english': 'As-Sabur',
    'desc_english': 'The Patient',
  },
 
};

// This includes the names and descriptions of Allah (Asmaul Husna) from 51 to 99.


example/lib/main.dart
import 'package:asmaulhusna/asmaulhusna.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
       
        primarySwatch: Colors.blue,
      ),
      home: const HomePage(),
    );
  }
}

class HomePage extends StatefulWidget {
  const HomePage({super.key});

  @override
  State<HomePage> createState() => HomePageState();
}

List<Text> nintynineNamesInEnglish() {
  List<Text> names = [];
  int i = 1;
  getEveryEnglishName().forEach((element) {
    names.add(Text('${i.toString()} - $element'));
    i++;
  });
  return names;
}

class HomePageState extends State<HomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
          child: Column(
        mainAxisSize: MainAxisSize.min,
        children: [
          Text(getArabicName(1)),
          Text(getEnglishName(1)),
          Text(getDescritption(1)),
          const SizedBox(
            height: 10,
          ),
          const Text('99 names'),
          const SizedBox(
            height: 10,
          ),
          Container(
              color: const Color.fromARGB(255, 231, 225, 166),
              height: 500,
              width: MediaQuery.of(context).size.width,
              child: SingleChildScrollView(
                  child: Column(
                children: [...nintynineNamesInEnglish()],
              )))
        ],
      )),
    );
  }
}
